packages/utils-debugging-protocol-common/src/debugging-protocol-connector.ts

```diff
@@ -22,26 +22,257 @@ import { Crdp } from 'chrome-remote-debug-protocol';
 
+enum RequestStatus {
+    willBeSent = 'willBeSent',
+    responseReceived = 'responseReceived',
+    loadingFinished = 'loadingFinished',
+    loadingFailed = 'loadingFailed'
+}
+
+type NetworkEvent = Crdp.Network.RequestWillBeSentEvent |
+    Crdp.Network.ResponseReceivedEvent |
+    Crdp.Network.LoadingFinishedEvent |
+    Crdp.Network.LoadingFailedEvent;
+
+let overrideInvalidCert: false;
+
+class RequestResponse {
+    /** The debugging protocol Network Client used to download the response body */
+    private _network: Crdp.NetworkClient;
+
+    /** The associated requestId for this network request. */
+    private _requestId: string;
+
+    private _status: RequestStatus;
+    /** The current status of the request. */
+    public get status() {
+        return this._status;
+    }
+
+    private _hops: string[] = [];
+    public get hops() {
+        return this._hops;
+    }
+
+    private _originalUrl: string;
+    /** The initial URL that started the request process before any redirects. */
+    public get originalUrl() {
+        return this._originalUrl;
+    }
+
+    /** The final URL after redirects. Could be the same as `originalUrl`. */
+    public get finalUrl(): string {
+        return this.responseReceived!.response.url;
+    }
+
+    private _willBeSent: Crdp.Network.RequestWillBeSentEvent | undefined;
+    /** The payload returned by the debugging protocol in the `RequestWillBeSent` event. */
+    public get willBeSent(): Crdp.Network.RequestWillBeSentEvent {
+        return this._willBeSent!;
+    }
+
+    private _responseReceived: Crdp.Network.ResponseReceivedEvent | undefined;
+    /** The payload returned by the debugging protocol in the `ResponseReceived` event. */
+    public get responseReceived() {
+        return this._responseReceived;
+    }
+
+    private _loadingFinished: Crdp.Network.LoadingFinishedEvent | undefined;
+    /** The payload returned by the debugging protocol in the `LoadingFinished` event. */
+    public get loadingFinished() {
+        return this._loadingFinished;
+    }
+
+    private _loadingFailed: Crdp.Network.LoadingFailedEvent | undefined;
+    /** The payload returned by the debugging protocol in the `LoadingFailed` event. */
+    public get loadingFailed() {
+        return this._loadingFailed;
+    }
+
+    private _responseBody: Crdp.Network.GetResponseBodyResponse | undefined;
+    public get responseBody() {
+        return this._responseBody;
+    }
+
+    private fetchContent(href: string, headers: HttpHeaders) {
+        const options = {
+            headers,
+            // we sync the ignore SSL error options with `request`. This is neeeded for local https tests
+            rejectUnauthorized: !overrideInvalidCert,
+            strictSSL: !overrideInvalidCert
+        };
+
+        const request: Requester = new Requester(options);
+
+        return request.get(href);
+    }
+
+    private _rawResponse: Buffer | undefined;
+    private getRawResponse() {
```

Comment:
don't you want this to be `private get rawResponse()` as the other properties??

Reply:
I wish I could, but it has some `async` stuff and the return type is `NetworkData` instead of `Buffer`

Reply:
Ahhh, ok! Maybe can you add the type to the function then. `private getRawResponse(): NetworkData`